python/GafferUI/CompoundEditor.py

```diff
@@ -1667,10 +1667,21 @@ def __showEditorFocusMenu( self, *unused ) :
 
 	def __addStandardItems( self, editor, m ) :
 
-		# @see CompoundEditor.nodeSetMenuSignal for details on the structure
-		# of this menu.
-
 		drivingEditor, _ = editor.getNodeSetDriver()
+		selection = editor.scriptNode().selection()
+
+		if len(selection) == 0 :
+			label = "Pin To Nothing"
+		elif len(selection) == 1 :
+			label = "Pin '%s'" % selection[0].getName()
+		else :
+			label = "Pin %d Nodes" % len(selection)
```

Comment:
Any merit in saying "Pin N _selected_ Nodes", to keep some of the clarity of the original message for this particular case?

Reply:
Improved [here](https://github.com/GafferHQ/gaffer/pull/4085/commits/4500624bd47c639437bd73fff152383d55ea5164).